pkg/compare/secret_key_reference.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package compare
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+)
+
+// SecretKeyReferenceEqual returns true if the supplied secret key references
+// are equal.
+func SecretKeyReferenceEqual(a, b *ackv1alpha1.SecretKeyReference) bool {
+	if HasNilDifference(a, b) {
+		return false
+	}
+	return a.Name == b.Name &&
+		a.Namespace == b.Namespace &&
+		a.Key == b.Key
+}
+
+// SliceStringPEqual returns true if the supplied slices of secret key reference
+// pointers contain the exact same elements. It will also return 2 slices containing
+// the added and removed elements. The comparison doesn't take into consideration
+// the order of elements.
+// Duplicated elements will not impact the behaviour of this function, meaning that
+// the function will always return only unique instances of the added/removed elements.
+//
+// TODO(a-hilaly): Ideally we will need to return another array containing secret
+// that got updated. To do that we will need to pass a k8s client to the newResourceDelta
+// function and use it here to retrieve/compare the secret values.
```

Comment:
I actually don't want to compare the values of the secrets. Just the keys/namespaces I think is good.